juju/juju
# Lxd first time failure

## Description of change

There are two parts to this change, the first one is the removal of the old mechanism for enabling the https listener. The second part is then implementing a retry mechanism where by we request a new local provider for the lxd client so that we can re-request the connection information correctly.

> Why is this change needed?

Juju bootstrap on bionic would fail on a lxd localhost, which caused the user to manually retry. The subsequent retry would work fine. 

## QA steps

> How do we verify that the change works?

Assume you have the source code, you can do the following:

On your local shell
```shell
lxc launch ubuntu:18.04 bionic
make go-build go-install
lxc file push ~/go/bin/juju* bionic/
lxc exec bionic bash
```
With in the lxc bionic shell 
```shell
lxd init --auto
mv /juju* ~
./juju bootstrap localhost test
```

## Documentation changes

> Does it affect current user workflow? CLI? API?

Bionic first time setup with lxd should work first time, instead of requiring a retry manually.

## Bug reference

> Does this change fix a bug? Please add a link to it.

- https://trello.com/c/bU1TseDj/258-bootstrapping-lxd-on-bionic-first-time-failure
- https://bugs.launchpad.net/juju/+bug/1771887
